tests/unit_tests/test_unit.py

```
@@ -1912,6 +1913,84 @@ def test_spectrum_id_mgf(mgf_small, tmp_path):
+        for msg in caplog.messages
+    )
+
+
+def test_shuffle_buffer_size_is_converted_to_batches(
+    mgf_small, tmp_path, monkeypatch
+):
+    """Test that shuffle_buffer_size is passed to the shuffler in batches."""
+    captured = {}
+
+    class FakeShufflerIterDataPipe:
+        def __init__(self, datapipe, buffer_size):
```

Ruff (0.15.7)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)